specreduce/tracing.py

```diff
@@ -177,25 +176,29 @@ class KosmosTrace(Trace):
         guess position. Useful for tracing faint sources if multiple
         traces are present, but potentially bad if the trace is
         substantially bent or warped. [default: None]
+    trace_model : one of `~astropy.modeling.polynomial.Chebyshev1D`,\
+            `~astropy.modeling.polynomial.Legendre1D`,\
+            `~astropy.modeling.polynomial.Polynomial1D`,\
+            or `~astropy.modeling.spline.Spline1D`, optional
+        The 1-D polynomial model used to fit the trace to the bins' peak
+        pixels. Spline1D models are fit with Astropy's
+        'SplineSmoothingFitter', while the other models are fit with the
+        'LevMarLSQFitter'. [default: ``models.Polynomial1D(degree=2)``]
     peak_method : string, optional
-        One of ``gaussian`` (default), ``centroid``, or ``max``.
-        gaussian: fits a gaussian to the window within each bin and
-        adopts the central value.  centroid: takes the centroid of the
-        window within in bin.  smooth_max: takes the position with the
-        maximum flux after smoothing over the window within each bin.
-
-    Improvements Needed
-    -------------------
-    1) switch to astropy models for Gaussian (done)
-    2) return info about trace width (?)
-    3) add re-fit trace functionality (or break off into other method)
-    4) add other interpolation modes besides spline, maybe via
-        specutils.manipulation methods?
+        One of ``gaussian``, ``centroid``, or ``max``.
+        ``gaussian``: Fits a gaussian to the window within each bin and
+        adopts the central value as the peak. May work best with fewer
+        bins on faint targets. (Based on the "kosmos" algorithm from
+        James Davenport's same-named repository.)
+        ``centroid``: Takes the centroid of the window within in bin.
+        ``max``: Saves the position with the maximum flux in each bin.
+        [default: ``max``]
     """
-    bins: int = 20
+    bins: int = None
     guess: float = None
     window: int = None
-    peak_method: str = 'gaussian'
+    trace_model: Model = field(default=models.Polynomial1D(degree=2))
```

Comment:
i would suggest setting the default degree to 1, i.e. a line. better to start with the assumption of no curvature and work up from there, if needed.